integration_test/src/test_runner/mod.rs

```diff
@@ -0,0 +1,327 @@
+    let password = env::var("ADL_TEST_PWD").expect("ADL_TEST_PWD is not set");
+    let host = env::var("ADL_TEST_HOST").expect("ADL_TEST_HOST is not set");
+    let auth_db = match env::var("ADL_TEST_AUTH_DB") {
+        Ok(val) => val,
+        Err(_) => "admin".to_string(), //Default auth db
+    };
+    let db = match env::var("ADL_TEST_DB") {
+        Ok(val) => val,
+        Err(_) => "INTEGRATION_TEST".to_string(), //Default driver name
+    };
+    let driver = match env::var("ADL_TEST_DRIVER") {
+        Ok(val) => val,
+        Err(_) => "ADL_ODBC_DRIVER".to_string(), //Default driver name
+    };
+
+    let connection_string = format!(
+        "Driver={};PWD={};USER={};SERVER={};AUTH_SRC={};Database={}",
+        driver, password, user_name, host, auth_db, db
+    );
+    let connection = ODBC_ENV
+        .connect_with_connection_string(&connection_string)
+        .unwrap();
+
+    assert!(!connection.is_dead().unwrap());
+    Ok(connection)
+}
+
+/// integration_test runs the query and function tests contained in the TEST_FILE_DIR directory
+#[test]
+#[ignore]
+pub fn integration_test() -> Result<(), Error> {
+    let paths = load_file_paths(PathBuf::from(TEST_FILE_DIR)).unwrap();
+    for path in paths {
+        let yaml = parse_test_file_yaml(&path).unwrap();
+
+        for test in yaml.tests {
+            match test.skip_reason {
+                Some(_) => continue,
+                None => {
+                    let connection = connect().unwrap();
+                    let test_result = match (&test.query, &test.function) {
+                        (Some(_), _) => run_query_test(&test, &connection),
+                        (None, Some(_)) => run_function_test(&test, &connection),
+                        (_, _) => {
+                            return Err(Error::MissingQueryOrFunction(format!(
+                                "{:?}",
+                                test.description
+                            )))
+                        }
+                    };
+                    drop(connection);
+                    return test_result;
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+fn run_query_test(entry: &TestEntry, conn: &Connection) -> Result<(), Error> {
+    let cursor = conn
+        .execute(entry.query.as_ref().unwrap(), ())
+        .unwrap()
+        .unwrap();
+    validate_result_set(entry, cursor);
+    Ok(())
+}
+
+fn str_or_null(value: &Value) -> *const u8 {
+    if value.is_null() {
+        null_mut()
+    } else {
+        odbc_api::handles::SqlText::new(value.as_str().expect("Unable to cast value as string"))
+            .ptr()
+    }
+}
+
+fn to_i16(value: &Value) -> i16 {
+    value.as_i64().expect("Unable to cast value as i64") as i16
+}
+
+fn check_array_length(array: &Vec<Value>, length: usize) {
+    if array.len() < length {
+        panic!(
+            "not enough values in array expected: {}, actual: {}",
+            length,
+            array.len()
+        )
```

Comment:
Should this instead return an error which can be propagated all the way up to `integration_test` and return like any other error, or panicking the behavior we prefer in this case?

Reply:
Okay, made the change to return an error instead.